eng/tools/rush-runner/index.js

```diff
@@ -240,16 +135,22 @@ function tryGetPkgRelativePath(absolutePath) {
     : absolutePath.substring(sdkDirectoryPathStartIndex);
 }
 
-const isReducedTestScopeEnabled = reducedDependencyTestMatrix[serviceDirs];
-if (isReducedTestScopeEnabled) {
-  // If a service is configured to have reduced test matrix then run rush for those reduced projects
-  console.log(`Found reduced test matrix configured for ${serviceDirs}.`);
-  packageNames.push(...reducedDependencyTestMatrix[serviceDirs]);
+let isReducedTestScopeEnabled = false;
+
+for (const dir of serviceDirs) {
+  if (reducedDependencyTestMatrix[dir]) {
+    isReducedTestScopeEnabled = true;
+    // If a service is configured to have reduced test matrix then run rush for those reduced projects
+    console.log(`Found reduced test matrix configured for ${serviceDirs}.`);
+    packageNames.push(...reducedDependencyTestMatrix[dir]);
+  }
 }
-const packagesWithDirection = getDirectionMappedPackages(packageNames);
+
+const packagesWithDirection = getDirectionMappedPackages(packageNames, actionComponents);
 const rushx_runner_path = path.join(baseDir, "common/scripts/install-run-rushx.js");
+let exitCode = 0;
 if (serviceDirs.length === 0) {
-  spawnNode(baseDir, "common/scripts/install-run-rush.js", action, ...rushParams);
+  exitCode = spawnNode(baseDir, "common/scripts/install-run-rush.js", action, ...rushParams);
```

Comment:
if this is a failing exit code should we exit early? What if this gets overridden by line 188 later?

Reply:
They are in the `else` branch but yes exit here should work.